klippy/extras/filament_switch_sensor.py

```diff
@@ -52,19 +53,24 @@ def _runout_event_handler(self, eventtime):
             pause_resume = self.printer.lookup_object('pause_resume')
             pause_resume.send_pause_command()
             pause_prefix = "PAUSE\n"
-        self._exec_gcode(pause_prefix, self.runout_gcode)
-        self.event_running = False
+            reactor = self.printer.get_reactor()
+            waketime = reactor.monotonic() + self.pause_delay
+            reactor.register_callback(
+                (lambda e, s=self, prefix=pause_prefix, gc=self.runout_gcode:
+                 s._exec_gcode(prefix, gc)), waketime)
```

Comment:
FYI, since `_runout_event_handler()` is already invoked in a callback handler, it's safe to just issue `self.printer.get_reactor().pause(eventtime + self.pause_delay)` here.

-Kevin

Reply:
Ah, good point.  I'll fix it up.